### ISSUE TITLE
"Unable to write diagnostic message to database" warning

### ISSUE THREAD

**Issue by arvindshmicrosoft** [github/codeql-action#2429]
I've noticed the below warning in recent runs of [this workflow](https://github.com/microsoft/SQLCallStackResolver/blob/main/.github/workflows/codeql-analyze.yml). 

```
Unable to write diagnostic message to database: Error: ENOENT: no such file or directory, open 'D:\a\_temp\codeql_databases\csharp\diagnostic\codeql-action\codeql-action-2024-08-16T19:58:32.722Z.json'
```

I'm hoping someone can shed some light on why this might be happening and more importantly if there is any implication on the analysis itself.

**Comment by nickfyson** [comment]
Thanks for your report, and we're looking into it to try and understand what's happening here.

In the meantime, commenting now just to reassure you that this won't be impacting the analysis itself! 👍

**Comment by aibaars** [comment]
@nickfyson The cause is likely that one of the sub-folders in `D:\a\_temp\codeql_databases\csharp\diagnostic\codeql-action` was not created.

**Comment by henrymercer** [comment]
Thanks for the report!  The CodeQL Action was trying to create a path containing characters that are invalid on Windows, specifically `:`.  This is fixed in the latest version v3.26.3.
